### PR TITLE
Implement size unit formatter and converter 84

### PR DESCRIPTION
**Pull request by Patti88**
## Summary by CodeRabbit

- **New Features**
  - Introduced a utility for formatting sizes into human-readable strings and converting between different measurement units, including byte and bit rate displays.
  - Enhanced support for parsing textual size representations and handling network rate formatting with improved precision.
  - Added a new enumeration for size units to facilitate better categorization and conversion.

- **Tests**
  - Added a comprehensive test suite to validate formatting, conversion, parsing accuracy, and edge case handling, ensuring reliable performance.

### CODE REVIEW

File: src/main/java/org/fungover/breeze/util/size/SizeFormatter.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            number = new BigDecimal(matcher.group(1));
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Invalid numeric value: " + matcher.group(1), e);
+        }
+
+        SizeUnit unit = parseUnit(matcher.group(3));
+        BigDecimal bytes = unit.toBytes(number);
+
+        if (bytes.compareTo(BigDecimal.valueOf(Long.MAX_VALUE)) > 0) {
+            throw new ArithmeticException("Size exceeds maximum Long value");
```
Comment (Stelle83) [inline src/main/java/org/fungover/breeze/util/size/SizeFormatter.java:61]: I recommend to provide more context in the message to help users understand the issue better.

for example:
throw new ArithmeticException("Parsed size exceeds maximum Long value: " + bytes + ". Ensure the size is within the acceptable range.");

File: src/main/java/org/fungover/breeze/util/size/SizeFormatter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                .replace(",", ".") + " " + unit.getSuffix();
+    }
+
+    private static SizeUnit parseUnit(String unitPart) {
+        for (SizeUnit unit : SizeUnit.values()) {
+            if (unit.getSuffix().equalsIgnoreCase(unitPart)) {
+                return unit;
+            }
+        }
+        throw new IllegalArgumentException("Unrecognized unit: " + unitPart);
```
Comment (Stelle83) [inline src/main/java/org/fungover/breeze/util/size/SizeFormatter.java:126]: same here, to be more clear for the user, would be good to add extra information for the user.
For example:
throw new IllegalArgumentException("Unrecognized unit: '" + unitPart + "'. Supported units are: B, KB, MB, GB, TB, KiB, MiB, GiB, TiB.");

File: src/main/java/org/fungover/breeze/util/size/SizeFormatter.java
Comment (Stelle83) [inline src/main/java/org/fungover/breeze/util/size/SizeFormatter.java:1]: Generally looks good!

Some comments:
*Code Comments: Add more detailed comments explaining the logic, especially within the autoFormat and parse methods.
*Unit Tests: Confirm comprehensive test coverage for all methods, including edge cases for parsing and formatting.

**Review by Stelle83 — CHANGES_REQUESTED** [review]
Generally looks good!

Some comments:
*Code Comments: Add more detailed comments explaining the logic, especially within the autoFormat and parse methods.
*Unit Tests: Confirm comprehensive test coverage for all methods, including edge cases for parsing and formatting.

**Comment by SeydaKinaci** [comment]
Your code is clear and understandable, but you could add Javadoc to make it even more readable. Also, I noticed that some of your commits are not signed, so it might be a good idea to fix that as well.

**Review by Stelle83 — DISMISSED** [review]
The changes improve the clarity and maintainability of the code by simplifying patterns, removing unused methods, and enhancing exception messages.

Overall, the refactoring and new tests look good and contribute positively to the codebase.

**Review by JesperSandklef — DISMISSED** [review]
Code looks good after the changes. It has improved the maintainability and looks cleaner with the unused methods removed. Approved

**Comment by JesperSandklef** [comment]
To get all your commits signed you can use rebase in intellij.

**Review by AfagMamedova — COMMENTED** [review]
Hej!
Det här var en smart funktion som gör det lättare att läsa storlekar, till exempel när man jobbar med filer.
Bra att det går att visa storlekar i t.ex. MB och GB.
Ser ut att fungera bra och koden är lätt att förstå.
Det skulle vara bra om man kan välja mellan 1000 och 1024 som bas.
Kanske lägga till ett felmeddelande om man skriver in något konstigt.
